File: MobiFon.Core/Entities/PropertyReservation.cs
using MobiFon.Core.Entities.Base;
using MobiFon.Core.Entities.Identity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MobiFon.Core.Entities
{
    public class PropertyReservation: BaseEntity
    {
        public Property Property { get; set; }
        public string ReservationNumber { get; set; }

        public int PropertyId { get; set; }
        public ApplicationUser Client { get; set; }
        public int ClientId { get; set; }
        public int NumberOfGuests { get; set; }
        public DateTime DateOfOccupancyStart { get; set; }  
        public DateTime DateOfOccupancyEnd { get; set; }  
        public int NumberOfDays { get; set; }
        public int NumberOfMonths { get; set; }
        public float TotalPrice { get; set; }
        public bool IsMonthly { get; set; }
        public bool IsDaily { get; set; }  
        public bool IsActive { get; set; }  
        public string? Description { get;set; }

    }
}

File: UI/propertease_client/lib/models/property_reservation.dart
import 'package:json_annotation/json_annotation.dart';

import 'application_user.dart';
import 'property.dart';

part 'property_reservation.g.dart';

@JsonSerializable()
class PropertyReservation {
  int? id;
  DateTime? createdAt;
  DateTime? modifiedAt;
  int? totalRecordsCount;
  bool? isDeleted;
  Property? property;
  int? propertyId;
  int? clientId;
  int? renterId;
  int? numberOfGuests;
  DateTime? dateOfOccupancyStart;
  DateTime? dateOfOccupancyEnd;
  int? numberOfDays;
  int? numberOfMonths;
  double? totalPrice;
  bool? isMonthly;
  bool? isDaily;
  bool? isActive;
  String? reservationNumber;
  ApplicationUser? client;
  ApplicationUser? renter;
  String? description;

  PropertyReservation({
    this.id,
    this.totalRecordsCount = 0,
    this.isDeleted = false,
    this.property,
    this.propertyId = 0,
    this.clientId = 1,
    this.numberOfGuests = 0,
    this.dateOfOccupancyStart,
    this.dateOfOccupancyEnd,
    this.numberOfDays = 0,
    this.numberOfMonths = 0,
    this.totalPrice = 0,
    this.isMonthly = true,
    this.isDaily = false,
    this.isActive = true,
    this.reservationNumber = '',
    this.client,
    this.renter,
    this.description,
  });

  factory PropertyReservation.fromJson(Map<String, dynamic> json) =>
      _$PropertyReservationFromJson(json);

  /// Connect the generated [_$PropertyReservationToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$PropertyReservationToJson(this);
}
